apps/portal/app/buckets/[address]/_components/bucket.tsx

```diff
@@ -120,15 +119,34 @@ export default function Bucket({ bucketAddress }: { bucketAddress: Address }) {
                   </Link>
                 </BreadcrumbLink>
               ) : (
-                displayAddress(bucketAddress)
+                <div className="text-foreground flex items-center gap-2 font-semibold">
+                  {displayAddress(bucketAddress)}
+                  <CopyButton
+                    value={bucketAddress}
+                    tooltip="Copy bucket ID"
+                    successMessage="Bucket ID copied to clipboard"
+                    className="opacity-40 hover:opacity-100"
+                  />
+                </div>
               )}
             </BreadcrumbItem>
             {!!pathParts.length && <BreadcrumbSeparator />}
             {pathParts.map((part, index) => (
               <Fragment key={`${index}-${part}`}>
                 <BreadcrumbItem>
                   {index === pathParts.length - 1 ? (
-                    part || "\u00A0\u00A0"
+                    <div className="text-foreground flex items-center gap-2 font-semibold">
+                      <File className="text-primary mr-1 size-4" />
+                      {part || "\u00A0\u00A0"}
+                      <CopyButton
+                        value={part}
```

Comment:
Was thinking about this more and I still think the copy button should copy the whole path. The primary use case of copying would be to use the value in the CLI. The CLI needs the entire path. Is there some other use case I'm not thinking of where just the last part of the path is needed?

Reply:
sure, was just going by the github norm. will switch

Reply:
there is a funny little quirk herein the UI, since you'd never want the bucket id copied in that path. so then if you get down to just the bucket it will flip to letting you copy the id